CONTRIBUTING.md

```diff
@@ -85,31 +100,48 @@ Binary output will be under each project's `bin\$(Configuration)\$(TargetFramewo
 
-To execute unit tests, just open `Test Explorer` and run tests from the explorer. Solution supports Live Unit Testing, there are interactive tests in the suite that are auto excluded from running with this feature. 
+To execute unit tests, just open `Test Explorer` and run tests from the explorer. Solution supports Live Unit Testing, there are interactive tests in the suite that are auto excluded from running with this feature.
+
+#### Visual Studio Code (experimental)
+
+> Note: Due to some usage of .NET Framework by source code, building and executing tests have been customized and features such as run and debug test out-of-box in the IDE won't execute correctly (see workaround below).
+
+Open  Visual Studio Code by navigating to the root directory of the repository in a PowerShell (or a command prompt) window and type `code .`, press `Enter`. Visual Studio Code should open with `POWERBI-POWERSHELL` in the Explorer pane. Settings, configurations, and tasks are auto-loaded by the files in the `.vscode` directory.
+
+To build, first-time run task `Restore` by pressing `Ctrl+P`, type `task Restore`, press `Enter`.
+After restoring packages you can execute build by pressing `Ctrl+Shift+B`.
+
+To run tests, highlight the name of the test with your cursor and press `Ctrl+P`, type `task DebugTest`, press `Enter`. If you don't want to build do `task DebugTest (No Build)` instead. Set your breakpoint in the test and press `F5` to start debugging by picking the dotnet process ID shown from `DebugTest`. Press `F5` once more to trigger the breakpoint.
 
 ### Creating a new Cmdlet and Unit Test
 
-1. Create a new class in a Module project and extend `PowerBICmdlet` from `Microsoft.PowerBI.Commands.Common` project and implement class.
-2. Add `[Cmdlet(CmdletVerb, CmdletName)]` attribute to class and provide `public const string` for `CmdletVerb` and `CmdletName`.
+1. Create a new class in a Module project.
+2. Either extend `PowerBIClientCmdlet` from `Microsoft.PowerBI.Common.Client` project or extend `PowerBICmdlet` from `Microsoft.PowerBI.Commands.Common` project if a client is not needed.
+3. Implement class and optionally add other interfaces from `Microsoft.PowerBI.Common.Abstractions` for common parameters.
+4. Add `[Cmdlet(CmdletVerb, CmdletName)]` attribute to class and provide `public const string` for `CmdletVerb` and `CmdletName`.
     * For `CmdletVerb`, pick from the System.Management.Automation.Verbs* classes.
-3. Optionally add `[OutputType(typeof(type))]` if your Cmdlet class if it writes an object to the output stream.
-4. Optionally add `[Alias("Get-Alias1", "Get-Alias2")]` to your Cmdlet class. 
-5. Add a test class to verify your cmdlet by adding (replacing with your new class name):
+5. Optionally add `[OutputType(typeof(type))]` if your Cmdlet class if it writes an object to the output stream.
```

Comment:
"if your Cmdlet class if it writes an object" -> "if your Cmdlet class writes an object"

Reply:
Ok